django/db/models/constraints.py

```diff
@@ -26,6 +26,9 @@ def clone(self):
         _, args, kwargs = self.deconstruct()
         return self.__class__(*args, **kwargs)
 
+    def supported(self, connection):
```

Comment:
Totally opened to a better name here.

Reply:
I have a suggestion. We previously added [`PostgresIndex.check_supported()`](https://github.com/django/django/blob/b33bfc383935cd26e19a2cf71d066ac6edd1425f/django/contrib/postgres/indexes.py#L32-L33) to handle checking support for various index types/flags. 

While that currently takes in `schema_editor`, we could change [this line](https://github.com/django/django/blob/b33bfc383935cd26e19a2cf71d066ac6edd1425f/django/contrib/postgres/indexes.py#L22) to pass `schema_editor.connection` and then we would have some consistency if you also adopted `def check_supported(self, connection):` here.

Reply:
Having said that, `PostgresIndex.check_supported()` doesn't return a boolean, it expects `NotSupportedError` to be raised as it allows checking for multiple conditions. We also don't handle those exceptions intentionally.

I'm not sure that is a deal breaker to making this look similar, however.